src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -371,16 +368,62 @@ private async ValueTask<EvaluationEvent> EvaluateFeature<TContext>(string featur
                     await sessionManager.SetAsync(evaluationEvent.FeatureDefinition.Name, evaluationEvent.Enabled).ConfigureAwait(false);
                 }
 
-                if (evaluationEvent.FeatureDefinition.Telemetry != null &&
-                    evaluationEvent.FeatureDefinition.Telemetry.Enabled)
+                // Only add an activity event if telemetry is enabled for the feature
+                if (telemetryEnabled)
                 {
-                    PublishTelemetry(evaluationEvent, cancellationToken);
+                    AddEvaluationActivityEvent(evaluationEvent);
                 }
             }
 
             return evaluationEvent;
         }
 
+        private void AddEvaluationActivityEvent(EvaluationEvent evaluationEvent)
+        {
+            // Break if there is no reason to emit an event
+            if (Activity.Current == null || !Activity.Current.IsAllDataRequested)
+            {
+                return;
+            }
+
+            ActivityTagsCollection tags = new ActivityTagsCollection
```

Comment:
This is a good place to use `var`, the type of `tags` is clear on the other side of the assignment.


```suggestion
            var tags = new ActivityTagsCollection
```

Reply:
You might actually want to use target-typed new here instead:
```suggestion
            ActivityTagsCollection tags = new()
```

Reply:
Even better.

Reply:
sounds good